### PR TITLE
Encourage using common keywords for plugins, configs, and formatters in documentation

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Related to #7838

> Is there anything in the PR that needs further explanation?

These keywords would help us find packages more easily.

Common keywords: `stylelint-plugin`, `stylelint-config`, `stylelint-formatter`

See:
- https://www.npmjs.com/search?q=keywords%3Astylelint-plugin
- https://www.npmjs.com/search?q=keywords%3Astylelint-config
- https://www.npmjs.com/search?q=keywords%3Astylelint-formatter

### CODE REVIEW

**Review by romainmenke — APPROVED** [review]
LGTM!

----

In our test framework for PostCSS plugins we also check `package.json` for the presence of `postcss-plugin` in keywords.

https://github.com/csstools/postcss-plugins/blob/e8809370ba61f406c3063b7a7566faf3223c4aae/packages/postcss-tape/src/index.ts#L98-L141

Is there an analogue to something like that for Stylelint?

We used to forget and overlook things like the keywords but having a test for it makes it easier.

File: docs/developer-guide/plugins.md [hunk 1]
```diff
@@ -362,4 +362,32 @@ To make a single module provide multiple rules, export an array of plugin object
...
-Use the `stylelint-plugin` keyword within your `package.json`.
+Use the `stylelint-plugin` keyword within your `package.json`. For example:
+
+```json
+{
+  "keywords": ["stylelint", "stylelint-plugin"]
+}
+```
+
+## Sharing configurations
```
Comment (Mouvedia) [inline docs/developer-guide/plugins.md:373]: We will eventually need a dedicated `docs/developer-guide/config.md` page for _Writing configurations_.

File: docs/developer-guide/plugins.md [hunk 2]
```diff
@@ -362,4 +362,32 @@ To make a single module provide multiple rules, export an array of plugin object
...
+
+You may want to create a shareable configuration for your plugins to provide a default ruleset. For example:
+
+```json
+{
+  "plugins": ["stylelint-plugin-foo"],
+  "rules": {
+    "foo/rule-1": true,
```
Comment (Mouvedia) [inline docs/developer-guide/plugins.md:381]: question
minor

Should be `stylelint-plugin-foo/rule-x`?

File: docs/developer-guide/plugins.md (hunk 1 above)
Comment (ybiquitous) [reply docs/developer-guide/plugins.md:373]: Probably yes. You can write it.

File: docs/developer-guide/plugins.md (hunk 2 above)
Comment (ybiquitous) [reply docs/developer-guide/plugins.md:381]: Actually, we don't need to keep consistency between a package name and a rule namespace. So, I think it's good enough as it is.

E.g.

https://github.com/stylelint/stylelint/blob/67c21375e12f695a3929bde27281d772dcdeaf93/docs/developer-guide/plugins.md?plain=1#L27-L35

**Comment by ybiquitous** [comment]
@romainmenke Thanks for the review!

> Is there an analogue to something like that for Stylelint?
> 
> We used to forget and overlook things like the keywords but having a test for it makes it easier.

As far as I know, there is no such thing, but that sounds great to me. 😃 
If we provide such a feature, [stylelint-test-rule-node](https://github.com/stylelint/stylelint-test-rule-node) and [jest-preset-stylelint](https://github.com/stylelint/jest-preset-stylelint) seems appropriate to me.

Or do you have a better idea about this?

**Comment by romainmenke** [comment]
I filed an issue here : https://github.com/stylelint/jest-preset-stylelint/issues/110 😄

**Pull request merged**